src/Offline/database.ts

```diff
@@ -108,16 +108,17 @@ export class Database implements IOfflineProvider {
         };
 
         var timeStampUsed = false;
-        var manifestURL = this._currentSceneUrl + ".manifest";
+        var manifestURL = new URL(this._currentSceneUrl)

```

Comment:
you will still need to check if `typeof URL !== 'undefined'`. Something like this:

```suggestion
        const manifestURL = typeof URL !== 'undefined' ? new URL(this._currentSceneUrl) : this._currentSceneUrl + ".manifest";
```

Of course the rest of the changes will need to react correctly to that as well. So maybe an `if( typeof URL !== 'undefined')` will work better

Reply:
Agreed. I've refactored into an internal function to avoid duplication.